apps/health_campaign_field_worker_app/lib/data/data_repository.dart

```diff
@@ -16,14 +17,79 @@ abstract class DataRepository<D extends DataModel, R extends DataModel> {
 abstract class RemoteRepository<D extends DataModel, R extends DataModel>
     extends DataRepository<D, R> {
   final Dio dio;
+  final String path;
+  final String entityName;
 
-  RemoteRepository(this.dio);
+  RemoteRepository(
+    this.dio, {
+    required this.path,
+    required this.entityName,
+  });
+
+  String get createPath => '$path/_create';
```

Comment:
We have the service-registry MDMS config where we can read the entire endpoint so that we don't have this fixed in code in case this naming convention changes